### PR TITLE
Activated accounts only

### PR DESCRIPTION
**Pull request by forgetso**
- [x] stops site keys from using the service if they aren't registered

### CODE REVIEW

File: packages/common/src/locales/en.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
 		"INVALID_SIGNATURE": "Invalid signature",
 		"NOT_IMPLEMENTED": "Not implemented",
 		"SITE_KEY_MISSING": "SITE KEY missing",
-		"ACCOUNT_NOT_FOUND": "Account not found"
+		"ACCOUNT_NOT_FOUND": "Account not found",
+		"INVALID_TIMESTAMP": "Invalid timestamp"
```
Comment (goastler) [inline packages/common/src/locales/en.json]: do we have the translations added for the new strings?

File: packages/common/src/logger.ts [hunk 2]
```diff
@@ -136,3 +136,9 @@ export class Loggable {
 		this.#logger = logger;
 	}
 }
+
+export const logError = (err: unknown, logger: Logger): void => {
+	logger.error(
+		typeof err === "object" && err ? ("stack" in err ? err.stack : err) : err,
```
Comment (goastler) [inline packages/common/src/logger.ts:142]: if the type of the err is object, won't this output "false"?

File: packages/procaptcha/src/modules/ProsopoCaptchaApi.ts [hunk 3]
```diff
@@ -60,6 +60,11 @@ export class ProsopoCaptchaApi implements ProsopoCaptchaApiInterface {
 				this.userAccount,
 				this.provider,
 			);
+
+			if (captchaChallenge[ApiParams.error]) {
```
Comment (goastler) [inline packages/procaptcha/src/modules/ProsopoCaptchaApi.ts:64]: why is this needed? It only skips the http->https code

File: packages/provider/src/api/captcha.ts [hunk 4]
```diff
@@ -167,12 +175,28 @@ export function prosopoRouter(env: ProviderEnvironment): Router {
 		try {
 			const { user, dapp } = GetPowCaptchaChallengeRequestBody.parse(req.body);
 
+			validateAddress(user, false, 42);
+			validateAddress(dapp, false, 42);
```
Comment (goastler) [inline packages/provider/src/api/captcha.ts:191]: duplicate?

**Review by goastler — CHANGES_REQUESTED** [review]

File: packages/provider/src/api/captcha.ts (hunk 4 above)
Comment (forgetso) [reply packages/provider/src/api/captcha.ts:191]: No, one is validating the `dapp` and the other is validating the `user`

File: packages/procaptcha/src/modules/ProsopoCaptchaApi.ts (hunk 3 above)
Comment (forgetso) [reply packages/procaptcha/src/modules/ProsopoCaptchaApi.ts:64]: because without the return the `captchas` key doesn't exist on the response so the following loop would fail. The API can now return a response with an `error` key, which, when encountered, is displayed in the widget as a red error message.

File: packages/common/src/logger.ts (hunk 2 above)
Comment (forgetso) [reply packages/common/src/logger.ts:142]: if the type of err is an object and err is not undefined
then if stack is in err, return err.stack
else return err
else return err

how does false come into this?

File: packages/common/src/locales/en.json (hunk 1 above)
Comment (forgetso) [reply packages/common/src/locales/en.json]: Yes, they were already present but I needed to move them into the `GENERAL` section which I've now done.

Comment (forgetso) [reply packages/common/src/locales/en.json]: I've also added a unit test to catch any missing translation keys

Comment (forgetso) [reply packages/common/src/locales/en.json]: https://github.com/prosopo/captcha/blob/5c4ace994c7208e8b25c73473c5d16487a8ef94a/packages/common/src/tests/locales.unit.test.ts

Comment (forgetso) [reply packages/common/src/locales/en.json]: This stuff is already in due to the locale PR.

File: packages/locale/.npmignore
Comment (goastler) [inline packages/locale/.npmignore:1]: remove pls

**Review by goastler — DISMISSED** [review]

File: packages/common/src/logger.ts (hunk 2 above)
Comment (goastler) [reply packages/common/src/logger.ts:142]:
```
typeof err === "object" && err ? ("stack" in err ? err.stack : err) : err
```
say err is not object, it becomes
```
false && err ? ("stack" in err ? err.stack : err) : err
```
false && <anything> is false, so
```
false
```

Comment (forgetso) [reply packages/common/src/logger.ts:142]: I'm ok with this. Most likely scenario is that error will be `undefined` but there's a very small chance of that considering its in try catch block.
